pkg/asset/installconfig/powervs/session.go

```diff
@@ -94,30 +87,36 @@ func getPISession() (*ibmpisession.IBMPISession, *UserCredentials, error) {
 			},
 		}, &apikey)
 		if err != nil {
-			return nil, nil, errors.New("Error saving the API_KEY variable")
+			return nil, errors.New("Error finding the API_KEY variable")
 		}
 	}
 
-	region = os.Getenv("IBMCLOUD_REGION")
 	// this can also be pulled from  ~/bluemix/config.json
-	if r2 := os.Getenv("IC_REGION"); len(r2) > 0 {
-		if len(region) > 0 && region != r2 {
-			return nil, nil, errors.New(fmt.Sprintf("conflicting values for IBM Cloud Region: IBMCLOUD_REGION: %s and IC_REGION: %s", region, r2))
-		}
-		if len(region) == 0 {
-			region = r2
+	var regionEnvVars = []string{"IBMCLOUD_REGION", "IC_REGION"}
+	region = getEnv(regionEnvVars)
+	if len(region) == 0 {
+		region, err = GetRegion()
+		if err != nil {
+			return nil, errors.New("Error finding the region")
 		}
 	}
 
-	if zone = os.Getenv("IBMCLOUD_ZONE"); len(zone) == 0 {
-		zone = region
+	var zoneEnvVars = []string{"IBMCLOUD_ZONE"}
+	zone = getEnv(zoneEnvVars)
+	if len(zone) == 0 {
+		zone, err = GetZone(region)
+		if err != nil {
+			return nil, errors.New("Error finding the zone")
+		}
 	}
 
-	// @TOOD: query if region is multi-zone? or just pass through err...
-	// @TODO: pass through debug?
 	s, err := ibmpisession.New(passwd, region, false, defSessionTimeout, id, zone)
-	uc := &UserCredentials{UserID: id, APIKey: apikey}
-	return s, uc, err
+	if err != nil {
+		return nil, err
+	}
+	s.IAMToken = apikey
```

Comment:
see above comment. just switch out the old passwd for apikey and then you can remove this line because New() will set IAMToken to our apikey